test/fuzz/transformation_load_test.cpp

```diff
@@ -293,6 +293,176 @@ TEST(TransformationLoadTest, BasicTest) {
   ASSERT_TRUE(IsEqual(env, after_transformation, context.get()));
 }
 
+TEST(TransformationLoadTest, DemoTestCase) {
+  /* (will remove this after PR ready)
+  I want to edit 'TransformationLoadTest' to work for both OpLoad and
+  OpAtomicLoad
+  ---------------------------------------------------------------
+  OpLoad:
+  %res_type_id = opCode(OpLoad) %res_id %ptr_id MemOperand(literal)(optional)
+  ----------------------------------------------------------------
+  OpAtomicLoad:
+  %res_type_id = opCode(OpAtomicLoad) %res_id %ptr_id %scope_id
+  %semantic_id(mask - 32-bitIntegerScalar)
+
+  load through Pointer using give semantics.
+
+  - **ptr_id** for atomic load must be one of this:
+  Uniform when, StorageBufferm, PhysicalStorageBuffer,
+  Workgroup, CrossWorkgroup, Generic AtomicCounter, Image, Function.
+  Type of result type must be the same of type the pointer point to.
```

Comment:
Note that the spec also says this, if the Shader capability is ues:

> Atomic access rules
>
>The pointers taken by atomic operation instructions are further restricted to not point into the Function storage class.

Thus your transformation should be designed to say that a transformation that tries to do an atomic load from a pointer with Function storage class is *not* allowed if the module uses the Shader capability.